nntrainer/compiler/ini_interpreter.cpp

```diff
@@ -39,10 +39,89 @@ namespace {
 class PlainLayer {};    /**< Plain Layer tag */
 class BackboneLayer {}; /**< Backbone Layer tag */
 
-template <typename T>
-static std::shared_ptr<Layer> section2layer(dictionary *ini,
+/**
+ * @brief convert section to list of string based properties
+ *
+ * @param ini ini handler
+ * @param sec_name section name
+ * @return std::vector<std::string> list of properties
+ */
+std::vector<std::string> section2properties(dictionary *ini,
```

Comment:
as `section2properties()` strongly overlaps with `parseProperties()` in ModelLoader, I recommend trying to keep only 1 implementation, either here or in ModelLoader, for easier maintenance. (you can make `ModelLoader::parseProperties()` as static public and use in `section2properties()` or vice versa)

Reply:
Since model_loader is including this header, I'll make model_loader to use this with some tweak in other PR.

Reply:
Then you might want to make this function more generic (taking the properties keys to be skipped).

Reply:
sure, that works too.

Reply:
Even, do we need model loader?

Reply:
Yes, model loader will call those functions altogether.

Reply:
I wonder that .. model loader ( which will be removed ) = section2properties + compiler?

Reply:
model loader will parses and loads graph(from serializers) + optimizers + dataset + model hyper parameters.